### ISSUE TITLE
MM: Learning Mode Switch not working right

### ISSUE THREAD

**Issue by TomRust** [lzim/teampsd#418]
Hi James,

It looks like the Learning Mode switch isn't quite working -- specifically when users select "Existing Patients RVI"

It looks like the sim is loading the wrong values...such that the model doesn't initialize in equilibrium. -- see screenshot below.  This issue only appears when users select "Existing Patients RVI," not with "New Patient Start Rate."

![image](https://user-images.githubusercontent.com/32961420/59568345-414d6f80-902e-11e9-9edd-e0f77c10d2d1.png)

![image](https://user-images.githubusercontent.com/32961420/59568407-216a7b80-902f-11e9-95ea-786345d54ad9.png)

**Comment by jamesmrollins** [comment]
@TomRust we will have to walk through this one together. I think we may need to address the model too.

**Comment by jamesmrollins** [comment]
@TomRust I am going to elevate this issue to Forio, as I cannot isolate a cause as to why the client version of the model is returning different values than the Forio instance of the model.  Will report back when I find something out.  May be an issue where the model is not receiving a user defined input of neither 1 nor 0, but not throwing an error.  I will ask Forio.

**Comment by jamesmrollins** [comment]
@TomRust @hirenp-waferwire  I sent the files to Forio so they could run them directly on the vensim model server and compare them to the results returned from the client version of vensim.  The returns were consistent with each other.  So I am not sure what the problem.  @hirenp-waferwire please take a look at the code as a potential source of the problem. Thanks, James

**Comment by jamesmrollins** [comment]
@TomRust @lzim upon investigation with Forio, the model instance running on the server is returning different results than the desktop version.  Ergo, we have to take this issue to Ventana for further investigation.  I will update as I learn more.

Furthermore, please be aware that the Care Coordination model is similarly affected by this learning mode switch problem.

**Comment by jamesmrollins** [comment]
@TomRust @lzim Fixed!!  I am going to run some tests, in test, then update production.  When this card appears on Done, we should be back to normal.

**Comment by jamesmrollins** [comment]
@TomRust @hirenp-waferwire  The problem turned out to be the improper use of a gaming variable.  I have changed the "Use Team Data for Start Rate or Return Visit Interval" variable from a gaming variable to a normal-constant variable.

- Evidently, when vensim enters into gaming mode, it initializes all the levels and rates. 
- To set these rates, it gathers all the constants presented in the model.
- Then it enters into gaming mode at step 0 (or in our case, -13).
- Then it runs 13 steps, or in our case from -13 to 0.
- When the "Use Team Data for Starting Rate or Return Visit Interval" is set as a gaming variable, the logic isn't introduced until after initialization, which in turn produces different results.

One can easily test this running either the cc or mm model in desktop Vensim, select "Use Team Data . . . " variable, _while running in gaming mode_ and using the formula pointer to change the logical condition.  It will give same "curvy" results we saw earlier, when stepped from -13 to 0.   

I tested the both models in Test and they are now fully working.  I have promoted these corrected models to PROD and have updated the version date to 2019_07_02 and uploaded into GitHub.
